### PR TITLE
[PAGOPA 667] add filter to psp

### PR DESCRIPTION
**Pull request by alessio-acitelli**
#### List of Changes

- add filters by name, status and psp code

#### Motivation and Context

- new filters required

#### How Has This Been Tested?

- Manual and junit test

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by alessio-acitelli** [comment]
update_code

**Comment by alessio-acitelli** [comment]
update_code

File: src/main/java/it/pagopa/pagopa/apiconfig/repository/PspRepository.java
Comment (jacopocarlini) [inline src/main/java/it/pagopa/pagopa/apiconfig/repository/PspRepository.java:1]: Can we increase readability by using specifications instead of queries?

**Comment by alessio-acitelli** [comment]
update_code

File: src/main/java/it/pagopa/pagopa/apiconfig/repository/PspRepository.java
Comment (alessio-acitelli) [reply src/main/java/it/pagopa/pagopa/apiconfig/repository/PspRepository.java:1]: done

**Review by jacopocarlini — APPROVED** [review]
Great Work 💪

**Pull request merged**